cpmpy/tools/explain/mus.py

```diff
@@ -108,6 +111,55 @@ def do_recursion(soft, hard, delta):
     return [dmap[a] for a in core]
 
 
+def optimal_mus(soft, hard=[], weights=1, solver="ortools", hs_solver="ortools"):
+    """
+        Find an optimal MUS according to a linear objective function.
+        Works by iteratively generating correction subsets and computing optimal hitting sets to those enumerated sets.
+        For better performance of the algorithm, use an incemental solver to compute the hitting sets such as Gurobi.
+
+        Assumption-based implementation for solvers that support s.solve(assumptions=...)
+        More naive version available as `optimal_mus_naive` to use with other solvers.
+
+        CPMpy implementation loosely based on the "SMUS" algorithm from
```

Comment:
is it? the OCUS paper is, but if it is only indirectly through the OCUS paper then I think citing that is enough?

We might want to more explicitly say that without overwriting 'weights' this will find the smallest mus... e.g. in the top of the doc?